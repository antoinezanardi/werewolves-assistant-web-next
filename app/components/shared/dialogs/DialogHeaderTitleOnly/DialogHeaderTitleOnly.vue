<template>
  <div
    id="dialog-header-title-only"
    class="w-full"
  >
    <h2 id="game-lobby-options-hub-header-title">
      <FontAwesomeIcon
        id="game-lobby-options-hub-header-title-icon"
        class="me-2"
        :class="iconClass"
        :icon="icon"
      />

      <span id="game-lobby-options-hub-header-title-text">
        {{ props.title }}
      </span>
    </h2>

    <PrimeVueDivider class="!my-2"/>
  </div>
</template>

<script setup lang="ts">
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import type { DialogHeaderTitleOnlyProps } from "~/components/shared/dialogs/DialogHeaderTitleOnly/dialog-header-title-only.types";

const props = defineProps<DialogHeaderTitleOnlyProps>();
</script>