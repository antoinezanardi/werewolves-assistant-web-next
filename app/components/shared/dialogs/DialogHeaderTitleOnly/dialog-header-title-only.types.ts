type DialogHeaderTitleOnlyProps = {
  title: string;
  icon: string;
  iconClass?: string;
};

export type { DialogHeaderTitleOnlyProps };