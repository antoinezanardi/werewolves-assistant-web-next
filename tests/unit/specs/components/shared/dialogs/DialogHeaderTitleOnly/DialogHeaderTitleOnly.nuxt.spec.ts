import type { mount } from "@vue/test-utils";

import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";
import type { DialogHeaderTitleOnlyProps } from "~/components/shared/dialogs/DialogHeaderTitleOnly/dialog-header-title-only.types";
import DialogHeaderTitleOnly from "~/components/shared/dialogs/DialogHeaderTitleOnly/DialogHeaderTitleOnly.vue";

describe("Dialog Header Title Only Component", () => {
  let wrapper: ReturnType<typeof mount<typeof DialogHeaderTitleOnly>>;
  const defaultProps: DialogHeaderTitleOnlyProps = {
    title: "Dialog Title",
    iconClass: "pi pi-info-circle",
  };

  async function mountDialogHeaderTitleOnlyComponent(options: ComponentMountingOptions<typeof DialogHeaderTitleOnly> = {}):
  Promise<ReturnType<typeof mount<typeof DialogHeaderTitleOnly>>> {
    return mountSuspendedComponent(DialogHeaderTitleOnly, {
      props: defaultProps,
      ...options,
    });
  }

  beforeEach(async() => {
    wrapper = await mountDialogHeaderTitleOnlyComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});