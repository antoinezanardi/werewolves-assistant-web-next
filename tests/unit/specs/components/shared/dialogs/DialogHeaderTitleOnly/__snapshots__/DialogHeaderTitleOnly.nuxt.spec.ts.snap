// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dialog Header Title Only Component > should match snapshot when rendered. 1`] = `
"<div id="dialog-header-title-only" class="w-full">
  <h2 id="game-lobby-options-hub-header-title">
    <font-awesome-icon-stub border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false" id="game-lobby-options-hub-header-title-icon" class="me-2 pi pi-info-circle"></font-awesome-icon-stub><span id="game-lobby-options-hub-header-title-text">Dialog Title</span>
  </h2>
  <divider-stub class="!my-2"></divider-stub>
</div>"
`;
